src/caches/sdirk_caches.jl

```diff
@@ -176,6 +176,64 @@ function alg_cache(alg::SDIRK2,u,rate_prototype,uEltypeNoUnits,uBottomEltypeNoUn
   SDIRK2Cache(u,uprev,fsalfirst,z₁,z₂,atmp,nlsolver)
 end
 
+struct SDIRK22ConstantCache{U,R,NR,Tab} <: OrdinaryDiffEqConstantCache
+  uhold::U
+  rhs::R
+  nl_rhs::NR
+  tab::Tab
+end
+
+function alg_cache(alg::SDIRK22,u,rate_prototype,uEltypeNoUnits,tTypeNoUnits,uBottomEltypeNoUnits,
+                   uprev,uprev2,f,t,dt,reltol,p,calck,::Type{Val{false}})
+  # 1-1/√2
+  a = 0.29289321881345254
```

Comment:
we should keep it using the analytical values and generate the tableau values using `eltype(u)`, so that way it support precision higher than Float64

Reply:
Ok, I will change to analytical values and use ```eltype(u)``` for the type.